File: backend/app/schemas/enterprise.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from app.models.enterprise import EnterpriseStatus, EnterpriseCategory, SalesStatus
from app.models.interaction import InteractionType


class InteractionBase(BaseModel):
    interaction_type: Optional[InteractionType] = InteractionType.OTHER
    date: Optional[datetime] = None
    description: str
    result: Optional[str] = None
    created_by: Optional[str] = None


class InteractionCreate(InteractionBase):
    pass


class InteractionResponse(InteractionBase):
    id: int
    enterprise_id: int
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class EnterpriseBase(BaseModel):
    name: str
    industry: Optional[str] = None
    employee_count: Optional[int] = None
    bank_penetration: Optional[float] = 0.0
    status: Optional[EnterpriseStatus] = EnterpriseStatus.PROSPECT
    category: Optional[EnterpriseCategory] = EnterpriseCategory.V
    score: Optional[int] = 0
    sales_status: Optional[SalesStatus] = SalesStatus.CONTACT
    inn: Optional[str] = None
    holding: Optional[str] = None
    locations: Optional[str] = None
    contact_person: Optional[str] = None
    contact_email: Optional[str] = None
    contact_phone: Optional[str] = None
    notes: Optional[str] = None
    manager: Optional[str] = None


class EnterpriseCreate(EnterpriseBase):
    pass


class EnterpriseUpdate(BaseModel):
    name: Optional[str] = None
    industry: Optional[str] = None
    employee_count: Optional[int] = None
    bank_penetration: Optional[float] = None
    status: Optional[EnterpriseStatus] = None
    category: Optional[EnterpriseCategory] = None
    score: Optional[int] = None
    sales_status: Optional[SalesStatus] = None
    inn: Optional[str] = None
    holding: Optional[str] = None
    locations: Optional[str] = None
    contact_person: Optional[str] = None
    contact_email: Optional[str] = None
    contact_phone: Optional[str] = None
    notes: Optional[str] = None
    manager: Optional[str] = None


class EnterpriseResponse(EnterpriseBase):
    id: int
    created_by_id: Optional[int] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    interactions: Optional[List[InteractionResponse]] = []

    class Config:
        from_attributes = True

File: frontend/src/types/enterprise.ts
// Enterprise types and interfaces

export interface ColumnConfig {
  id: string
  label: string
  visible: boolean
  sortable: boolean
}

export interface TableSettings {
  columns: ColumnConfig[]
  sortBy: string | null
  sortOrder: 'asc' | 'desc'
}

export interface Interaction {
  id: number
  enterprise_id: number
  interaction_type: 'call' | 'meeting' | 'email' | 'presentation' | 'contract' | 'other'
  date: string
  description: string
  result: string | null
  created_by: string | null
  created_at: string
}

export interface Enterprise {
  id: number
  name: string
  industry: string | null
  employee_count: number | null
  bank_penetration: number | null
  status: 'prospect' | 'negotiation' | 'pilot' | 'active' | 'inactive'
  category: 'A' | 'B' | 'V' | 'G'
  score: number
  sales_status: 'planned' | 'contact' | 'negotiation' | 'contract' | 'launched'
  inn: string | null
  holding: string | null
  locations: string | null
  contact_person: string | null
  contact_email: string | null
  contact_phone: string | null
  notes: string | null
  manager: string | null
  created_by_id: number | null
  created_at: string | null
  updated_at: string | null
  interactions: Interaction[]
}

export interface ImportPreview {
  columns: string[]
  sample_data: Record<string, unknown>[]
  suggested_mapping: Record<string, string | null>
  available_fields: Record<string, { label: string; required: boolean }>
  total_rows: number
  mapping_method?: 'llm' | 'fallback'
}

export interface NewInteraction {
  interaction_type: Interaction['interaction_type']
  description: string
  result: string
  created_by: string
}
